src/psyclone/core/access_info.py

```diff
@@ -419,29 +420,54 @@ def next_location(self):
 
     def add_access(self, signature, access_type, node, component_indices=None):
         '''Adds access information for the variable with the given signature.
+        If the `component_indices` parameter is not an instance of
+        ComponentIndices, it is used to construct an instance. Therefore it
+        can be None, a list or a list of lists of PSyIR nodes.
 
         :param signature: the signature of the variable.
         :type signature: :py:class:`psyclone.core.Signature`
         :param access_type: the type of access (READ, WRITE, ...)
         :type access_type: :py:class:`psyclone.core.access_type.AccessType`
         :param node: Node in PSyIR in which the access happens.
         :type node: :py:class:`psyclone.psyir.nodes.Node` instance
-        :param component_indices: list of lists of indices used in the \
-            access, one list for each component. None if the variable is \
-            not an array. Defaults to None, which is then converted to [[]].
-        :type component_indices: list of lists of \
-            :py:class:`psyclone.psyir.nodes.Node`
+        :param component_indices: index information for the access.
+        :type component_indices: \
+            :py:class:`psyclone.core.component_indices.ComponentIndices`, or \
+            any other type that can be used to construct a ComponentIndices \
+            instance (None, list or lists of \
+            :py:class:`psyclone.psyir.nodes.Node`)
 
         '''
-        # TODO 1268: Allow for more intuitive arguments, e.g. for simple
-        # arrays accept a non-nested list.
         if not isinstance(signature, Signature):
             raise InternalError("Got '{0}' of type '{1}' but expected it to "
                                 "be of type psyclone.core.Signature."
                                 .format(signature, type(signature).__name__))
 
-        if component_indices is None:
-            component_indices = [[]]
+        # To make it easier for the user, we allow to implicitly create the
+        # component indices instance here:
+        if not isinstance(component_indices, ComponentIndices):
+            # Handle some convenient cases:
+            # 1. Add the right number of [] if component_indices is None:
+            if component_indices is None:
+                component_indices = [[]] * len(signature)
+            elif isinstance(component_indices, list):
+                # 2. If the argument is a simple list (not a list of lists),
```

Comment:
While I agree, we can't do this atm, there is still some code that provides strings. I've opened #1320 with a detailed list of where this happens, including the additional consistency checks that should be added. 
While it might be straightforward to wrap the strings in `Literal(...)` or `Reference(...)` calls, I would actually prefer to look at the way the PSyIR nodes are now created for the variables introduces by PSyclone, and use the instances that already exist (instead of recreating say a reference, that's never in the actual tree). It might lead to some code simplification.
Note I did not add #TODO 1320 in the sources, the list of files is in the ticket, and I don't want to touch all these other files for an issue that is not really related to this PR.

Reply:
That all sounds fine, thanks.

Reply:
Done.

Reply:
That's much clearer now, thanks.